fastval.go

```diff
@@ -304,10 +310,22 @@ func (val FastVal) compareBoolean(other FastVal) int {
 }
 
 func (val FastVal) compareStrings(other FastVal) int {
-	// TODO: Improve string comparisons to avoid casting or converting
-	escVal, _ := val.AsJsonString()
-	escOval, _ := other.AsJsonString()
-	return strings.Compare(string(escVal.sliceData), string(escOval.sliceData))
+	switch other.dataType {
+	case RegexValue:
+		regex := other.GetRegex()
```

Comment:
I think that we need to have a different method used for performing "like" matching.  Otherwise the logic will get a bit confusing when doing some forms of matching I think?  LIKE matching is an explicit operation in the grammar and expression tree anyways.

Reply:
I think that Equality checking and matching are two discreet operations.  A new op should be added for the matcher, which would then call the alternative method.

Reply:
Understood